ctr/tests/mod.rs

```diff
@@ -33,3 +33,43 @@ fn test_from_cipher() {
         assert_eq!(&buf, &ciphertext);
     }
 }
+
+#[test]
+fn compare_to_openssl_with_over_64bit_counter() {
```

Comment:
Wondering if there are some other checks we should do for counter wrapping behavior, possibly ones from an RFC...

Reply:
> [...] possibly ones from an RFC...

Did you mean these https://tools.ietf.org/html/rfc3686#section-6 or something else? Either way, sounds like a great idea. I am not at all confident that this patch will work all around as I am not a cryptographic engineer.

Perhaps the original POC (testing against openssl) should be integrated to be used manually (under some feature flag, as the openssl dependency can be difficult), and/or doing some small number of blocks side-by-side on CI?

Reply:
Yes, something like that, if those actually have the desired behavior (I was hoping RFC 5297 might have something like it but alas it does not)

For what it's worth, here are the counter wrap tests I wrote for AES-GCM-SIV based on RFC 8452 (except it uses a 32-bit little endian counter):

https://github.com/RustCrypto/AEADs/blob/master/aes-gcm-siv/tests/ctr_wrap.rs